patches/server/0005-Paper-config-files.patch

```diff
@@ -3373,6 +3376,54 @@ index 0000000000000000000000000000000000000000..ef0e834c164b0ccc1a61b349348e6799
 +        });
 +    }
 +}
+diff --git a/src/main/java/io/papermc/paper/configuration/transformation/global/versioned/V29_LogIPs.java b/src/main/java/io/papermc/paper/configuration/transformation/global/versioned/V29_LogIPs.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..77c3370a3b87347d7be5bce7d682acd0ab6574d2
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/configuration/transformation/global/versioned/V29_LogIPs.java
+@@ -0,0 +1,42 @@
++package io.papermc.paper.configuration.transformation.global.versioned;
++
++import net.minecraft.server.MinecraftServer;
++import net.minecraft.server.dedicated.DedicatedServer;
++import org.checkerframework.checker.nullness.qual.Nullable;
++import org.spongepowered.configurate.ConfigurateException;
++import org.spongepowered.configurate.ConfigurationNode;
++import org.spongepowered.configurate.NodePath;
++import org.spongepowered.configurate.transformation.ConfigurationTransformation;
++import org.spongepowered.configurate.transformation.TransformAction;
++
++import static org.spongepowered.configurate.NodePath.path;
++
++public class V29_LogIPs implements TransformAction {
++
++    private static final int VERSION = 29;
++    private static final NodePath PATH = path("logging", "log-player-ip-addresses");
++    private static final V29_LogIPs INSTANCE = new V29_LogIPs();
++
++    private V29_LogIPs() {
++    }
++
++    public static void apply(ConfigurationTransformation.VersionedBuilder builder) {
++        builder.addVersion(VERSION, ConfigurationTransformation.builder().addAction(PATH, INSTANCE).build());
++    }
++
++    @Nullable
++    @Override
++    public Object[] visitPath(final NodePath path, final ConfigurationNode value) throws ConfigurateException {
++        DedicatedServer server = ((DedicatedServer) MinecraftServer.getServer());
++
++        boolean val = value.getBoolean(server.settings.getProperties().logIPs);
++        server.settings.getProperties().logIPs = val;
++        server.settings.getProperties().properties.setProperty("log-ips", String.valueOf(val));
++        server.settings.forceSave();
```

Comment:
Is there some way we can use DedicatedServerSettings#update to avoid needing to remove the final field? Like just change the property and then reconstruct the properties? Maybe its not actually better, but if it is, it removes the need to remove the final

Reply:
Nope, those are specifically mutable values, changing the logIPs value to also be a mutable value would involve a lot more diff and work than just stripping final;
<img width="1202" alt="image" src="https://github.com/PaperMC/Paper/assets/44026893/c53d322f-c7e1-49c1-a134-621cbe76d52b">

Reply:
Yeah, I saw that, but you shouldn't need to change it to be a MutableValue, just do something like 
```java
        this.settings.update(properties -> {
            properties.properties.setProperty("logIps", newValue);
            return properties.reload(whatever)
        })
```